### PR TITLE
Add method for deep copy of an entry

### LINKED ISSUE #16: Memory VDR store should return a deep copy of the DID document and metadata
Instead of pointer to the internal struct. Now you can alter the DID Document without going through `Update()`.

Suggestion: provide `Copy()` func on `DIDDocument`

### PR DESCRIPTION
**Pull request by gerardsn**
Fixes #16

### CODE REVIEW

File: vdr/store/memory.go [hunk 1]
```diff
@@ -109,26 +109,41 @@ func (m *memory) Resolve(id did.DID, metadata *vdr.ResolveMetadata) (*did.Docume
 	}
 
 	// return a deep copy
-	doc := &did.Document{}
-	docMetadata := &vdr.DocumentMetadata{}
+	copyEntry, err := deepCopy(entry)
```
Comment (reinkrul) [inline vdr/store/memory.go:112]: not sure if we want to have deepCopy() return an error? A memory copy should alway be able to succeed imo. I believe @woutslakhorst checked why/when marshal/unmarshal can fail

File: vdr/store/memory.go [hunk 2]
```diff
@@ -109,26 +109,41 @@ func (m *memory) Resolve(id did.DID, metadata *vdr.ResolveMetadata) (*did.Docume
...
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &copyEntry.document, &copyEntry.metadata, nil
+}
 
+// deepCopy returns a deep copy of a memoryEntry
+func deepCopy(entry *memoryEntry) (*memoryEntry, error) {
```
Comment (reinkrul) [inline vdr/store/memory.go:121]: I thought it was going to be a function in `go-did`?

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: vdr/store/memory.go (hunk 2 above)
Comment (woutslakhorst) [reply vdr/store/memory.go:121]: go-did can do a proper memory based deepCopy. For now it's fixed here. An issue still has to be made on go-did.

File: vdr/store/memory.go (hunk 1 above)
Comment (woutslakhorst) [reply vdr/store/memory.go:112]: when implemented in go-did yes. Now it's just on par with what it was but a bit more separated.

**Comment by woutslakhorst** [comment]
Failing tests on circleCI!

File: go.mod [hunk 3]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/labstack/echo/v4 v4.1.17
 	github.com/lestrrat-go/jwx v1.0.7
 	github.com/magiconair/properties v1.8.4
-	github.com/nuts-foundation/go-did v0.0.0-20210125091337-21da6ad69a82
```
Comment (stevenvegt) [inline go.mod:14]: Why the update of go-did?

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**